sklearn/preprocessing/tests/test_common.py

```diff
@@ -74,7 +82,9 @@ def test_missing_value_handling(est, func, support_sparse, strictly_positive):
         # train only on non-NaN
         est.fit(_get_valid_samples_by_column(X_train, i))
         # check transforming with NaN works even when training without NaN
-        Xt_col = est.transform(X_test[:, [i]])
+        with pytest.warns(None) as records:
+            Xt_col = est.transform(X_test[:, [i]])
+        assert len(records) == 0
```

Comment:
This new `test_common.py` assertion breaks with the PowerTransform that complains with all-nan columns.

I am not sure if we should raise this warning or not (maybe not at transform time) but this should be consistent across all the transformers.

Reply:
I made a push. The reason was on the `np.nanmin(X)` to check that the matrix was strictly positive. This case will return a full NaN matrix as well so everything will be fine (or at least the problem is forwarded to the next step in the pipeline).